### PR TITLE
support updated cargo component

### LINKED ISSUE #14: Build error: did you forget to run cargo component build?
It appears while previously booting, upon a fresh clone of the repo I'm getting the error.. 

```
john@raptor:~/code/wasm2$ make run
  Generating bindings for graph (/home/john/code/wasm2/crates/graph/src/bindings.rs)
   Compiling graph v0.1.0 (/home/john/code/wasm2/crates/graph)
error: bindings file `/home/john/code/wasm2/target/bindings/graph/bindings.rs` does not exist
       
       did you forget to run `cargo component build`? (https://github.com/bytecodealliance/cargo-component)
 --> crates/graph/src/lib.rs:1:1
  |
1 | cargo_component_bindings::generate!();
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `cargo_component_bindings::generate` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: use of undeclared crate or module `bindings`
 --> crates/graph/src/lib.rs:4:5
  |
4 | use bindings::exports::wasmbuilder_app::graph::provider::{
  |     ^^^^^^^^ use of undeclared crate or module `bindings`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `graph` (lib) due to 2 previous errors
make: *** [Makefile:12: build] Error 101
```

I've ran `make setup` a few times, make a third clone of the repo, still the same issue...

### PR DESCRIPTION
**Pull request by James-Mart**
Bindings generator in the graph crate is no longer valid with the updated version of cargo-component, which [no longer supports the bindgen macro](https://github.com/bytecodealliance/cargo-component/releases/tag/v0.7.0).

I believe this resolves #14.

### CODE REVIEW

File: .yarnrc.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+nodeLinker: node-modules
+
+yarnPath: .yarn/releases/yarn-4.1.0.cjs
```
Comment (peterhuene) [inline .yarnrc.yml:3]: Forgive my ignorance re: yarn, but is this basically committing a specific version of the yarn executable into the repo for consistency?

Is `./yarn/install-state.gz` something that would get updated when the lock file updates?

File: yarn.lock
Comment (peterhuene) [inline yarn.lock:1]: Is the huge diff in `yarn.lock` the result of using an updated `yarn` (lots of quoted entries in the new file)?

**Review by peterhuene — APPROVED** [review]
Looks good re: the Rust changes; just some questions regarding `yarn`.

File: .yarnrc.yml (hunk 1 above)
Comment (James-Mart) [reply .yarnrc.yml:3]: Correct, so everyone uses the same version, which will help avoid conflicts in the lock file, etc.

Re: `install-state.gz`, I don't consider myself a front-end guy either, so I had to look this up and learned that it shouldn't be checked in, actually, so I've removed it. See the changes in `.gitignore`

File: yarn.lock
Comment (James-Mart) [reply yarn.lock:1]: Yep

**Review by peterhuene — APPROVED** [review]

**Comment by peterhuene** [comment]
Thanks for the fix!

**Pull request merged**